kvstore/vault.go

```diff
@@ -152,15 +157,35 @@ func valueMap(value string) map[string]interface{} {
 
 // Vault API
 
-func (c *VaultKeyValueStore) GetData(key string, version int) (map[string]interface{}, error) {
+func (c *VaultKeyValueStore) beforeRequest(ctx context.Context, path string, action string) context.Context {
+	ctx, span := c.Tracer.Start(ctx, "vault request")
+	span.SetAttributes(attribute.Key("path").String(path), attribute.Key("action").String(action))
```

Comment:
~ Maybe renaming `action` to `method`? That's the pattern for HTTP instrumentation:
https://ui.honeycomb.io/circleci/datasets/distributed-traces/result/A1rDKpcxnij